### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by codymccoy**

### CODE REVIEW

File: ViewModel/ImpactAreaScenario/Results/PerformanceAEPVM.cs [hunk 1]
```diff
@@ -39,13 +39,13 @@ private void LoadData(metrics.ImpactAreaScenarioResults iasResult, List<Threshol
...
 
                     //get the histogram data
-                    Statistics.Histograms.ThreadsafeInlineHistogram histogramOfAEPs = performanceResults.GetAssurance("AEP").AssuranceHistogram;
-                    int[] binCounts = histogramOfAEPs.BinCounts;
-                    double binWidth = histogramOfAEPs.BinWidth;
-                    double min = histogramOfAEPs.Min;
-                    double[] binsAsDoubles = binCounts.Select(x => (double)x).ToArray();
-                    HistogramData2D _data = new HistogramData2D(binWidth, min, binsAsDoubles, "Chart", "Series", "X Data", "YData");
-                    HistogramData.Add(threshold, _data);
+                    //Statistics.Histograms.ThreadsafeInlineHistogram histogramOfAEPs = performanceResults.GetAssurance("AEP").AssuranceHistogram;
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Results/PerformanceAEPVM.cs:42]: Dead code?

File: ViewModel/ImpactAreaScenario/Results/PerformanceAssuranceOfThresholdVM.cs [hunk 2]
```diff
@@ -26,11 +26,11 @@ private void LoadData(metrics.ImpactAreaScenarioResults iasResult, List<Threshol
                     double exceedanceProb = 1.0 - xVal;
                     try
                     {
-                        Threshold threshold = iasResult.PerformanceByThresholds.GetThreshold(thresholdKey);
-                        SystemPerformanceResults projectPerformanceResults = threshold.SystemPerformanceResults;
-                        double yVal = projectPerformanceResults.GetAssurance("STAGE", exceedanceProb).AssuranceHistogram.CDF(threshold.ThresholdValue);
+                        //Threshold threshold = iasResult.PerformanceByThresholds.GetThreshold(thresholdKey);
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Results/PerformanceAssuranceOfThresholdVM.cs:29]: Dead code?

File: ViewModel/Tabs/TabController.cs [hunk 3]
```diff
@@ -67,28 +67,28 @@ public static TabController Instance
...
-                    indexToRemove = i;
-                    break;
-                }
-            }
-            if( indexToRemove != -1)
-            {
-                Tabs.RemoveAt(indexToRemove);
-            }
-        }
+        ///// <summary>
```
Comment (RyanM-RMA) [inline ViewModel/Tabs/TabController.cs:70]: Dead code

**Review by RyanM-RMA — CHANGES_REQUESTED** [review]
Got some minor changes to resolve here

File: ViewModel/Alternatives/AlternativeElement.cs [hunk 4]
```diff
@@ -106,14 +106,28 @@ private void AddActions()
...
+            if (vr.IsValid)
             {
-                MessageBox.Show(vr.ErrorMessage, "Invalid Setup", MessageBoxButton.OK, MessageBoxImage.Exclamation);
+                // Run calculations. waiting for hec to put the new model in.
+                //grab the result objects off the ias elements and run the calculation.
+                IASElementSet[] iASElems = GetElementsFromID();
+                IASElementSet firstElem = iASElems[0];
+                IASElementSet secondElem = iASElems[1];
+
+                //scenarios.Scenario scenario1 = new scenarios.Scenario(firstElem.AnalysisYear, );
```
[the remark this answers is not in the document]
Comment (codymccoy) [reply ViewModel/Alternatives/AlternativeElement.cs:119]: im leaving commented out. I put a comment at the top of that method.

File: ViewModel/ImpactAreaScenario/Results/PerformanceAEPVM.cs (hunk 1 above)
Comment (codymccoy) [reply ViewModel/ImpactAreaScenario/Results/PerformanceAEPVM.cs:44]: I am leaving it in. I commented why it is commented.

File: View/ImpactArea/ImpactAreaImporter.xaml.cs [hunk 5]
```diff
@@ -41,5 +35,12 @@ private void UserControl_DataContextChanged(object sender, DependencyPropertyCha
                 row_SelectUniqueName.Height = new GridLength(35);
             }
         }
+
+        private void cmb_Path_SelectionMade(string fullpath, string filename)
```
Comment (RyanM-RMA) [inline View/ImpactArea/ImpactAreaImporter.xaml.cs:39]: rename to appropriate naming standards (PascalCase)

File: View/Utilities/NullableDoubleTextBox.xaml.cs [hunk 6]
```diff
@@ -42,7 +42,12 @@ private void MyPreviewTextInput(object sender, TextCompositionEventArgs e)
         {
             e.Handled = true;
             string value = CombineText(e.Text);
+
+            //Regex regex = new Regex("[^0-9.-]+");
```
Comment (RyanM-RMA) [inline View/Utilities/NullableDoubleTextBox.xaml.cs:46]: Zombie code?  Why is this around?

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs [hunk 7]
```diff
@@ -417,49 +417,59 @@ private List<StageDamageCurve> GetStageDamageCurves()
...
             {
-                msg = "A Rating Curve is required when using a frequency relationship.";
+                vr.AddErrorMessage("A Rating Curve is required when using a frequency relationship.");
             }
-            return msg;
+            return vr;
+
         }
-        private string IsStageDamageValid()
+        private FdaValidationResult IsStageDamageValid()
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:442]: Follow the same naming conventions as GetPlotValidationResults.  Is implies bools are returned.

File: ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs [hunk 8]
```diff
@@ -417,49 +417,59 @@ private List<StageDamageCurve> GetStageDamageCurves()
...
+            FdaValidationResult vr = new FdaValidationResult();
+            if(SelectedFrequencyElement.ChildElement == null)
+            {
+                vr.AddErrorMessage("A Frequency Relationship is required.");
+            }
+            return vr;
         }
 
-        private string IsRatingCurveValid()
+        private FdaValidationResult IsRatingCurveValid()
```
Comment (RyanM-RMA) [inline ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs:430]: Follow the same naming conventions as GetPlotValidationResults.  Is implies bools are returned.

File: ViewModel/WaterSurfaceElevation/WaterSurfaceElevationImporterVM.cs [hunk 9]
```diff
@@ -185,59 +185,62 @@ private FdaValidationResult ValidateVRTFile(List<string> vrtFiles, string direct
...
         public void FileSelected(string fullpath)
         {
-            if (IsCreatingNewElement)
+            if (fullpath != null && !fullpath.Equals(SelectedPath))
             {
-                FdaValidationResult importResult = new FdaValidationResult();
-                ListOfRows.Clear();
-                //clear out any already existing rows
-                if (!Directory.Exists(fullpath))
+                if (IsCreatingNewElement)
```
Comment (RyanM-RMA) [inline ViewModel/WaterSurfaceElevation/WaterSurfaceElevationImporterVM.cs:190]: You can combine this with the one above.

**Pull request merged**